content/articles/how-to-use-a-local-dns-resolver-to-redirect-all-dns-requests/index.md

```diff
@@ -0,0 +1,115 @@
+
+However, I've put it in the list if you forget to set it to `DNS` based on your `Destination port range` decision.
+
+If you select `Destination/Invert`, the rule will match any traffic originating from the IoT network and not use the local IOT gateway IP address to resolve DNS requests. Put another way, and this rule will be activated on any DNS traffic meant for DNS servers situated on the Internet or even on other areas of your network. OPNsense configures each network's interface address as the gateway/DNS address by default.
+
+192.168.30.0-192.168.30.255 is the network IP address range for an IoT network with a 192.168.30.1 interface IP address, which is the gateway and DNS server addresses for that network. As a result, the DNS traffic to 192.168.30.1 will be allowed, but all other traffic will be redirected to 127.0.0.1. So there is no need to turn this address, so the redirect rule is not triggered. All interface addresses, such as 192.168.30.1, will resolve to the local Unbound DNS resolver, the localhost address of 127.0.0.1, which is why the redirect target IP address is set to the localhost address.
+
+You may want to go to the firewall rules page for the interface where you've put a NAT port forward rule after clicking `Save`. Firewall > Rules > IoT, for example. A rule was automatically inserted at the end of the list, as you can see. Depending on the other rules you have defined, this auto-generated rule may need to be moved to the top of the list. The other rule(s) should not interfere with the correct functioning of the rule.
+
+However, if you have many networks, this rule must be repeated for each network where DNS requests are redirected. Continued repetition could become dull and irritating. Also, if you decide to change the law in the future, it will require extra upkeep. Fortunately, there's a technique to create a single rule for numerous interfaces that's simple.
+
+### Configuration of multiple Interfaces
+You can construct a firewall group to apply a single rule to numerous interfaces. Using firewall groups, you can arrange several interfaces into a single group and write rules that apply to all of the interfaces in the group at once. When it comes to the ruling hierarchy, group rules take precedence over a set of interface rules. DNS redirection rules are likely to be executed before interface rules. Therefore you shouldn't have any problems. It is important to keep in mind the rule processing order while developing group rules for multiple interfaces.
+
+Go to the `Firewall > Groups` page to create a rule group. Then select "Add" from the drop-down menu.
+
+![Add](/engineering-education/how-to-use-a-local-dns-resolver-to-redirect-all-dns-requests/image-03)

```

Comment:
This image is incorrect it refers to the `Firewall > NAT` section instead of the `Firewall > Groups` section

It seems that image 4 is the one that's supposed to be here

Reply:
Hello, @adrianmurage. I had made the mistake of duplicating image-02 and image-03, whereas image-03 was supposed to have an `Add` button as rectified using this  [image-03](https://github.com/Daniel-muriithi/engineering-education/blob/how-to-use-a-local-dns-resolver-to-redirect-all-dns-requests/content/articles/how-to-use-a-local-dns-resolver-to-redirect-all-dns-requests/image-03.png), which I have added with the `Add` button.

For the [image-04](https://github.com/Daniel-muriithi/engineering-education/blob/how-to-use-a-local-dns-resolver-to-redirect-all-dns-requests/content/articles/how-to-use-a-local-dns-resolver-to-redirect-all-dns-requests/image-04.png) it is correctly named. As from the text above, assigning the firewall group  `Name` and selecting the `interfaces/networks` to which the DNS request is redirected is shown in image-04.